Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,471 @@
+                    }
+                }
+            } else {
+                // Process a single parallel force. Subtract 1 from index b/c
+                // we use 0 for the non-parallel forces.
+                const auto& impl =
+                    m_enabledParallelForces->getElt(threadIndex-1)->getImpl();
+                    if (!impl.dependsOnlyOnPositions()) {
+                        impl.calcForce(*m_state,
+                                m_rigidBodyForcesLocalStatic.upd(), m_particleForcesLocalStatic.upd(),
+                                m_mobilityForcesLocalStatic.upd());
+                    }
+            }
+            break;
+        }
+    }
+    
+    //Once a thread has finished it's force calculations, we add in the thread's
+    //contribution into the cached force arrays in the State
+    void finish() override {
+        *m_rigidBodyForces += m_rigidBodyForcesLocalStatic.get();
+        *m_particleForces += m_particleForcesLocalStatic.get();
+        *m_mobilityForces += m_mobilityForcesLocalStatic.get();
+
+        if (m_mode == CachedAndNonCached) {
+            *m_rigidBodyForceCache += m_rigidBodyForceCacheLocalStatic.get();
+            *m_particleForceCache += m_particleForceCacheLocalStatic.get();
+            *m_mobilityForceCache += m_mobilityForceCacheLocalStatic.get();
+        }
+    }
+private:
+    Mode m_mode;
+
+    ReferencePtr<const State> m_state;
+
+    // Constant state-cache variables for the enabled parallel and non-parallel
+    // forces.
+    ReferencePtr<const Array_<Force*>> m_enabledNonParallelForces;
+    ReferencePtr<const Array_<Force*>> m_enabledParallelForces;
+
+    // ReferencePtrs that point to caches in the state; We eventually add our
+    // thread-local result to these vectors.
+    ReferencePtr<Vector_<SpatialVec>> m_rigidBodyForces;
+    ReferencePtr<Vector_<Vec3>> m_particleForces;
+    ReferencePtr<Vector> m_mobilityForces;
+
+    ReferencePtr<Vector_<SpatialVec>> m_rigidBodyForceCache;
+    ReferencePtr<Vector_<Vec3>> m_particleForceCache;
+    ReferencePtr<Vector> m_mobilityForceCache;
+    
+    // These variables are local to a thread. They are set to their default
+    // value when the threads are spawned. We use them to keep track of each
+    // thread's contribution that we will later add in to the final state cache.
+    ThreadLocal<Vector_<SpatialVec>> m_rigidBodyForcesLocalStatic;
+    ThreadLocal<Vector_<Vec3>> m_particleForcesLocalStatic;
+    ThreadLocal<Vector> m_mobilityForcesLocalStatic;
+
+    ThreadLocal<Vector_<SpatialVec>> m_rigidBodyForceCacheLocalStatic;
+    ThreadLocal<Vector_<Vec3>> m_particleForceCacheLocalStatic;
+    ThreadLocal<Vector> m_mobilityForceCacheLocalStatic;
+};
+
+//local declarations of static member variables
+static ThreadLocal<Vector_<SpatialVec>> m_rigidBodyForcesLocalStatic();
+static ThreadLocal<Vector_<Vec3>> m_particleForcesLocalStatic();
+static ThreadLocal<Vector> m_mobilityForcesLocalStatic();
```

Comment:
And these are not doing anything because they aren't class member variables -- they are just new file-scope static variables that are coincidentally named like member variables. Should be:
```c++
/*static*/ ThreadLocal<Vector> CalcForcesParallelTask::m_mobilityForcesLocalStatic();
```
etc. Note that the word "static" is not allowed here. I usually leave it in as a comment as shown.